internal/controller/warehouses/warehouses.go

```diff
@@ -629,6 +633,36 @@ func validateDiscoveredArtifacts(
 		message = strings.Join(parts[:len(parts)-1], ", ") + ", and " + parts[len(parts)-1]
 	}
 
+	ok, err := freightCreationFilterSatisfied(ctx, warehouse, artifacts)
+	if err != nil {
+		conditions.Set(
+			newStatus,
+			&metav1.Condition{
+				Type:   kargoapi.ConditionTypeHealthy,
+				Status: metav1.ConditionFalse,
+				Reason: "FreightCreationFilterExpressionError",
+				Message: fmt.Sprintf(
+					"failed to evaluate freight creation filter expression: %s", err.Error(),
+				),
+				ObservedGeneration: warehouse.GetGeneration(),
+			},
+		)
+		return false
+	}
+	if !ok {
+		conditions.Set(
+			newStatus,
+			&metav1.Condition{
+				Type:               kargoapi.ConditionTypeHealthy,
+				Status:             metav1.ConditionTrue,
+				Reason:             "FreightCreationFilterNotSatisfied",
+				Message:            "freight creation filter expression not satisfied; skipping freight creation",
+				ObservedGeneration: warehouse.GetGeneration(),
+			},
+		)
+		return false
+	}
```

Comment:
I'm clear on why we're checking the filters in this function that seems only intended to validate discovery results instead of somewhere around line +289 where there is the existing check for creating Freight or not based on Freight creation policy. These filters are just new criteria for that same decision.